.github/workflows/build-test-deploy.yml

```diff
@@ -260,48 +262,85 @@ jobs:
       run: NO_APT_INSTALL=1 ./tests/test.sh --ignore=tests/test_executable.py
 
   build-container-aarch64:
-    if: startsWith(github.ref, 'refs/tags/')
-    needs: build-executable-aarch64
-    runs-on: ["self-hosted", "ARM64"]
+    runs-on:
+      - self-hosted
+      - public
+      - ARM64
+
     steps:
-      - name: Checkout Code
-        uses: actions/checkout@v3
-        with:
-          fetch-depth: 0
-          submodules: true
+    - name: Checkout Code
+      uses: actions/checkout@v3
+      with:
+        submodules: true
 
-      - name: Set up runner requirements
-        run: scripts/setup_runner_requirements.sh
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v2
 
-      - name: Get and verify tag value
-        run: ./scripts/verify_tag.sh
+    - name: Login to DockerHub
+      uses: docker/login-action@v1
+      with:
+        username: ${{ secrets.DOCKERHUB_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_TOKEN }}
 
-      - name: Login to DockerHub
-        uses: docker/login-action@v1
-        with:
-          username: ${{ secrets.DOCKERHUB_USERNAME }}
-          password: ${{ secrets.DOCKERHUB_TOKEN }}
+    # the tests need the gprofiler image built (from Dockerfile). I run it separately here, because "docker build" prints the build logs
+    # more nicely. the tests will then be able to use the built image.
+    - name: Build gProfiler image
+      run: ./scripts/build_aarch64_container.sh -t gprofiler_aarch64 --output type=docker
```

Comment:
Since https://github.com/Granulate/gprofiler/pull/790, the gProfiler image is a blank image that contains the executable.

Please see the job `build-container-x64` - it has
```
    needs:
      - build-executable-x64
```
to make sure the CI runs it after the exe is built, and then it uses the same binary exe built earlier to put it into the image, as done here:
```
      run: ./scripts/build_x86_64_container.sh --skip-exe-build --build-arg EXE_PATH=output/gprofiler_x86_64 -t gprofiler_x86_64
```

Please do the same here - by not passing `--skip-exe-build` you are rebuilding it unnecessarily, and plus, in any case, we'd like them to use the same file exactly.